packages/snaps-utils/src/structs.ts

```diff
@@ -547,3 +549,22 @@ export function mergeStructs(...structs: Struct<any>[]): Struct<any> {
     },
   });
 }
+
+/**
+ * Refine a struct to be a non-empty record.
+ *
+ * @param Key - The struct for the record key.
+ * @param Value - The struct for the record value.
+ * @returns The refined struct.
+ */
+export function nonEmptyRecord<Key extends string, Value>(
```

Comment:
We have a couple of places we likely want to use this in the SDK, so I would move it there